resources/entrypoints/main.sh

```diff
@@ -40,13 +57,38 @@ else
 
 	set -- "$@" --nodaemon
 	if [ "$1" == "ndb_mgmd" ]; then
-		echo "[Entrypoint] Starting ndb_mgmd"
-		set -- "$@" -f $RONDB_DATA_DIR/config.ini --configdir=$RONDB_DATA_DIR/log
+		echo "[entrypoints/main.sh] Starting ndb_mgmd"
+		set -- "$@" -f "$RONDB_DATA_DIR/config.ini" --configdir="$RONDB_DATA_DIR/log"
 	elif [ "$1" == "ndbmtd" ]; then
-		echo "[Entrypoint] Starting ndbmtd"
+
+		# ndbmtd has several hard-coded file creation modes that cannot
+		# be configured. Permissions can be removed from such hard-coded
+		# modes using umask, but there is no way to add permissions to
+		# them. As a workaround, this is a very hacky background process
+		# that every 5 seconds makes sure that the group's permissions
+		# equal the owner's.
+		while true; do
+			# Find all files owned by the current user, print their
+			# modestring and path, null-terminated.
+			find /srv/hops/mysql-cluster -user "$USER" -printf '%m %p\0' |
+			# Remove all null-terminated items that begin with two
+			# equal characters (where the group's permissions
+			# already equals the user's) and then remove the
+			# modestring.
+			sed -zr '/^(.)\1/d; s/^... //;' |
+			# xargs: Run chmod with an efficient number of file
+			# arguments to correct the group's permissions.
+			xargs -r0 chmod -cf g=u ||
+			# Make sure the process does not exit due to some
+			# failure.
+			true
+			sleep 5
+		done &
```

Comment:
Btw, what happens if the ndbmtd crashes? Could be that the last files are not deletable anymore? Which PID does this even use if the ndbmtd is started with `exec`?

Reply:
> Btw, what happens if the ndbmtd crashes? Could be that the last files are not deletable anymore?

Yes, if ndbmtd exits within 5 seconds after creating a new file we're in trouble as you say. I'll fix it, but that means I can't use `exec` for `ndbmtd`.

> Which PID does this even use if the ndbmtd is started with `exec`?

The `&` after `done` creaes a sub shell, so the background bash process becomes a child of `ndbmtd` which looks kinda funny:
```
docker exec ndbd_1 bash -c 'ps -fuxa; grep -H Umask /proc/*/status'
USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root        1477  0.0  0.0   4360  3176 ?        Ss   07:29   0:00 bash -c ps -fuxa; grep -H Umask /proc/*/status
root        1483  0.0  0.0   7060  1584 ?        R    07:29   0:00  \_ ps -fuxa
root           1  0.0  0.0   4360  3300 ?        Ss   07:02   0:00 /bin/bash ./docker_entrypoints/rondb_standalone/entrypoint.sh ndbmtd --ndb-nodeid=1 --initial --ndb-connectstring=mgmd_1:1186
root          30  0.0  0.0   7236  4348 ?        S    07:02   0:00 sudo -E -u mysql /srv/hops/docker_entrypoints/rondb_standalone/main.sh ndbmtd --ndb-nodeid=1 --initial --ndb-connectstring=mgmd_1:1186
mysql         31  0.0  0.0  29636 10216 ?        S    07:02   0:00  \_ ndbmtd --ndb-nodeid=1 --initial --ndb-connectstring=mgmd_1:1186 --nodaemon
mysql         34  0.0  0.0   4360  1776 ?        S    07:02   0:00      \_ /bin/bash /srv/hops/docker_entrypoints/rondb_standalone/main.sh ndbmtd --ndb-nodeid=1 --initial --ndb-connectstring=mgmd_1:1186
mysql       1476  0.0  0.0   2788  1008 ?        S    07:29   0:00      |   \_ sleep 5
mysql         40 29.3  8.0 6409960 5309384 ?     Sl   07:02   8:06      \_ ndbmtd --ndb-nodeid=1 --initial --ndb-connectstring=mgmd_1:1186 --nodaemon
/proc/1/status:Umask:   0022
/proc/1476/status:Umask:        0002
/proc/1477/status:Umask:        0022
/proc/30/status:Umask:  0022
/proc/31/status:Umask:  0002
/proc/34/status:Umask:  0002
/proc/40/status:Umask:  0002
/proc/self/status:Umask:        0022
/proc/thread-self/status:Umask: 0022

```

Reply:
But of course this will also not be the case after the fix, since I no longer use `exec`.